android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -2126,6 +2116,21 @@ public static void setGlobeKeyState(GlobeKeyState state) {
     globeKeyState = state;
   }
 
+  private static void toggleSuggestionBanner(String languageID, boolean result1, boolean result2) {
```

Comment:
I don't really like the `result1`, `result2` names -- are these related to in-app vs system keyboard?

Reply:
Yes. they're the respective statuses of `prepareKeyboardSwitch()` or `setKeyboard()`

`result1` for in-app keyboard
`result2` for system keyboard

Reply:
Is there a reason they can't be combined into a single boolean?  How would `false, false` and `true, true` be handled if not?

Reply:
Are you recommending something like this?
```
if (result) {
  InAppKeyboard.setLayoutParams(getKeyboardLayoutParams());
  SystemKeyboard.setLayoutParams(getKeyboardLayoutParams());
}
```

Reply:
More like

```
if (result) {
  InAppKeyboard.setLayoutParams(getKeyboardLayoutParams());
} else {
  SystemKeyboard.setLayoutParams(getKeyboardLayoutParams());
}
```

Reply:
Hmm - that doesn't handle the (often) scenario where both InApp and SystemKeyboard need to update `setLayoutParams()`.

Reply:
> Hmm - that doesn't handle the (often) scenario where both InApp and SystemKeyboard need to update `setLayoutParams()`.

That sounds quite different from how the system keyboard works on iOS - are in-app and system keyboards both part of the same process on Android?  I'm curious how that's a scenario at all, let alone a frequent one.  (On iOS, it's only ever one or the other.  They're not in the same process.)

Reply:
In Keyman Engine for Android, all keyboard switch calls in KMManager are applied to both in-app and system at the same time.